doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,198 @@
 
-       ~~~
-       $ sudo apt-get install libbz2-dev libbz2-1.0 zlib1g zlib1g-dev
-       $ wget http://sourceforge.net/projects/boost/files/boost/1.58.0/boost_1_58_0.tar.gz
-       $ tar -zxf boost_1_58_0.tar.gz
-       $ cd boost_1_58_0/
-       $ ./bootstrap.sh
-       $ sudo ./b2 install
-       ~~~
+        #include <stdio.h>
+        #include <kaa/kaa.h>
+        #include <kaa/platform/kaa_client.h>
+        #include <kaa/kaa_error.h>
 
-4. Install the [**AvroC++**](http://avro.apache.org/docs/1.7.6/api/cpp/html/index.html) library manually:
 
-   ~~~
-   $ wget http://archive.apache.org/dist/avro/avro-1.7.5/cpp/avro-cpp-1.7.5.tar.gz
-   $ tar -zxf avro-cpp-1.7.7.tar.gz
-   $ cd avro-cpp-1.7.5/
-   $ cmake -G "Unix Makefiles"
-   $ sudo make install
-   ~~~
+        #define KAA_APP_PRINT_TIMES 5
 
-5. Install the [**Botan**](http://botan.randombit.net/) library by executing the following command:
+        static void dummy_function(void* context)
+        {
+            for (size_t i=1; i<KAA_APP_PRINT_TIMES; i++) {
+                printf("Hello, I am a Kaa Application!\n");
+            }
+            kaa_client_stop(context);
+        }
 
-   ~~~
-   $ wget https://github.com/randombit/botan/archive/1.11.28.tar.gz
-   $ tar -zxf 1.11.28.tar.gz
-   $ cd botan-1.11.28/
-   $ ./configure.py
-   $ sudo make install
-   $ sudo ln -s /usr/local/include/botan-1.11/botan /usr/local/include/botan
-   ~~~
+        int main()
+        {
+            kaa_client_t* kaa_client = NULL;
+            kaa_error_t error = kaa_client_create(&kaa_client, NULL);
+            if (error) {
+                return -1;
+            }
 
-6. Install the **[SQLite](https://www.sqlite.org/index.html)** library by executing the following command:
-    1. For automatic installation, execute the following commands (tested on Ubuntu 14.04):
+            error = kaa_client_start(kaa_client, &dummy_function, (void *)kaa_client, 0);
 
-       ~~~
-       $ sudo apt-get install libsqlite3-0 libsqlite3-dev
-       ~~~
+            kaa_client_destroy(kaa_client);
 
-    2. For manual installation, refer to the following example:
+            if (error) {
+                return -1;
+            }
 
-       ~~~
-       $ wget https://www.sqlite.org/2015/sqlite-autoconf-3081002.tar.gz
-       $ tar -zxf sqlite-autoconf-3081002.tar.gz
-       $ cd sqlite-autoconf-3081002/
-       $ ./configure
-       $ sudo make install
-       ~~~
+            return 0;
+        }
 
-**NOTE:** Instead of manually installing all required components and libraries, you can follow [the quick way to build C/C++ endpoint SDK](#Linux-Quickway).  (only applicable for x86\_64 platform build)
+    The application code is simple: application just prints *KAA_APP_PRINT_TIMES* times the string *"Hello, I am a Kaa Application!\n"*. That's it. So, now we have the following file structure:
 
-## Compile C++ endpoint SDK
+        - my-kaa-application
```

Comment:
Do not post intermediate directory structure. This isn't something that user looks for. Reference build final build directory for clarity before build step.